### PR TITLE
samples: espi: update for gpio_dt_spec & DEVICE_DT_GET_ONE

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: samples/drivers/espi/src/main.c [hunk 1]
```diff
@@ -1212,62 +1203,52 @@ int espi_test(void)
...
+		return -ENODEV;
 	}
 #endif
 
 	LOG_INF("Hello eSPI test %s", CONFIG_BOARD);
 
 #if DT_NODE_HAS_STATUS(BRD_PWR_NODE, okay)
-	ret = gpio_pin_configure(pwrgd_dev, BRD_PWR_PWRGD_PIN,
-				 GPIO_INPUT | GPIO_ACTIVE_HIGH);
+	ret = gpio_pin_configure_dt(&pwrgd_gpio, GPIO_INPUT);
```
Comment (gmarull) [inline samples/drivers/espi/src/main.c:1237]: is `GPIO_ACTIVE_HIGH` reflected in DT now?

File: samples/drivers/espi/src/main.c [hunk 2]
```diff
@@ -1212,62 +1203,52 @@ int espi_test(void)
...
 	if (ret) {
-		LOG_ERR("Unable to configure %d:%d",
-			BRD_PWR_PWRGD_PIN, ret);
+		LOG_ERR("Unable to configure %d:%d", pwrgd_gpio.pin, ret);
 		return ret;
 	}
 
-	ret = gpio_pin_configure(rsm_dev, BRD_PWR_RSMRST_PIN,
-				 GPIO_OUTPUT | GPIO_ACTIVE_HIGH);
+	ret = gpio_pin_configure_dt(&rsm_gpio, GPIO_OUTPUT);
```
Comment (gmarull) [inline samples/drivers/espi/src/main.c:1243]: ditto

File: samples/drivers/espi/src/main.c (hunk 1 above)
Comment (galak) [reply samples/drivers/espi/src/main.c:1237]: Yes, it was already in the DTS.

Comment (galak) [reply samples/drivers/espi/src/main.c:1237]: Yes, it was already in the DTS.

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**